src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -90,6 +90,14 @@ internal static class ConstraintsHelper
             }
 
             diagnosticsBuilder.Free();
+
+            if (bounds is null && canUseLightweightTypeConstraintBinding)
```

Comment:
>if (bounds is null && canUseLightweightTypeConstraintBinding) [](start = 12, length = 61)

Would we need this additional logic if we never cached "light-bound" constraints on a symbol?